src/test/kotlin/io/github/mojira/arisa/infrastructure/IntegrationTest.kt

```diff
@@ -0,0 +1,39 @@
+package io.github.mojira.arisa.infrastructure
+
+import com.uchuhimo.konf.Config
+import io.github.mojira.arisa.infrastructure.config.Arisa
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.booleans.shouldBeTrue
+import io.kotest.matchers.maps.shouldContainKey
+import java.io.File
+
+class IntegrationTest : StringSpec({
+    "should be able to read the JSON correctly" {
+        val config = Config { addSpec(Arisa) }
+            .from.map.flat(
+                mapOf(
+                    "arisa.credentials.username" to "test",
+                    "arisa.credentials.password" to "test",
+                    "arisa.credentials.dandelionToken" to "test"
+                )
+            )
+            .from.json.watchFile("arisa.json")
+            .from.env()
+            .from.systemProperties()
+
+        config.containsRequired().shouldBeTrue()
```

Comment:
I think it would make sense to require almost all then.
Currently most things are still optional, but the module doesn't really do anything without them

Reply:
We should make things that the module breaks without required